endpoints/playlist.py

```diff
@@ -7,70 +7,77 @@
+    @room_required
+    def post(self, request, room):
+        """
+        Update playlist
+        ---
+        serializer: PlaylistSerializer
+        """
+        pk = request.data.get('pk')
+        if pk is None:
+            return Response('POST action needs a pk parameter', status=status.HTTP_400_BAD_REQUEST)
         try:
-            target = PlaylistTrack.objects.get(pk=int(target), room=room)
+            playlistTrack = PlaylistTrack.objects.get(pk=pk, room=room)
         except PlaylistTrack.DoesNotExist:
-            return Response("Can't find this playlistTrack as target.", status=status.HTTP_404_NOT_FOUND)
+            return Response("Can't find this playlistTrack.", status=status.HTTP_404_NOT_FOUND)
 
-    if target is not None:
-        getattr(playlistTrack, action)(target)
-    else:
-        getattr(playlistTrack, action)()
+        action = request.data.get('action')
+        if action not in PlaylistTrack.ACTIONS:
+            return Response('Action can only be: "%s"' % '" or "'.join(PlaylistTrack.ACTIONS), status=status.HTTP_400_BAD_REQUEST)
 
-    message = {
-        'action': 'playlistTrack_updated',
-        'playlistTracks': PlaylistSerializer(room.playlist.all(), many=True).data
-    }
-    room.send_message(message)
-    return Response(PlaylistSerializer(room.playlist.all(), many=True).data, status=status.HTTP_200_OK)
+        target = request.data.get('target')
+        if action in {'above', 'below'}:
+            if target is None:
+                return Response('"%s" action needs a target parameter' % action, status=status.HTTP_400_BAD_REQUEST)
+            try:
+                target = PlaylistTrack.objects.get(pk=int(target), room=room)
+            except PlaylistTrack.DoesNotExist:
+                return Response("Can't find this playlistTrack as target.", status=status.HTTP_404_NOT_FOUND)
 
+        if target is not None:
+            getattr(playlistTrack, action)(target)
+        else:
+            getattr(playlistTrack, action)()
 
-@api_view(['DELETE'])
-@room_required
-def delete(request, room, pk, format=None):
-    """
-    Delete music from playlist
-    ---
-    serializer: PlaylistSerializer
-    """
-    try:
-        PlaylistTrack.objects.get(pk=pk, room=room).delete()
-    except PlaylistTrack.DoesNotExist:
-        return Response("Can't find this playlistTrack.", status=status.HTTP_404_NOT_FOUND)
+        message = {
+            'action': 'playlistTrack_updated',
+            'playlistTracks': PlaylistSerializer(room.playlist.all(), many=True).data
+        }
+        room.send_message(message)
+        return Response(PlaylistSerializer(room.playlist.all(), many=True).data, status=status.HTTP_200_OK)
+
+    @room_required
+    def delete(self, request, room):
+        """
+        Delete music from playlist
+        ---
+        serializer: PlaylistSerializer
+        """
+        pk = request.data.get('pk')
```

Comment:
I go for 2 distinct decorator ([because this](http://stackoverflow.com/questions/739654/how-to-make-a-chain-of-function-decorators-in-python)), code will go to mind blow if I try do create a decorator with args... (or I MY mind is blowing while reading the explanations about decorators...)